src/plugins/analysis/qemu_exec/code/qemu_exec.py

```diff
@@ -0,0 +1,301 @@
+            return None
+        finally:
+            binary_service.shutdown()
+
+
+class AnalysisPlugin(AnalysisBasePlugin):
+
+    NAME = 'qemu_exec'
+    DESCRIPTION = 'test binaries for executability in QEMU and display help if available'
+    VERSION = '0.3.0'
+    DEPENDENCIES = ['file_type']
+    FILE_TYPES = ['application/x-executable']
+
+    FACT_EXTRACTION_FOLDER_NAME = 'fact_extracted'
+
+    arch_to_bin_dict = OrderedDict([
+        ('Alpha', ['alpha']),
+
+        ('aarch64', ['aarch64']),
+        ('ARM', ['aarch64', 'arm', 'armeb']),
+
+        ('MIPS32', ['mipsel', 'mips', 'mipsn32', 'mipsn32el']),
+        ('MIPS64', ['mips64', 'mips64el']),
+        ('MIPS', ['mipsel', 'mips', 'mips64', 'mips64el', 'mipsn32', 'mipsn32el']),
+
+        ('S/390', ['s390x']),
+
+        ('80386', ['i386']),
+        ('80486', ['x86_64', 'i386']),
+        ('x86', ['x86_64', 'i386']),
+
+        ('SPARC', ['sparc', 'sparc32plus', 'sparc64']),
+
+        ('PowerPC', ['ppc', 'ppc64', 'ppc64abi32', 'ppc64le']),
+        ('PPC', ['ppc', 'ppc64', 'ppc64abi32', 'ppc64le']),
+
+        ('Renesas SH', ['sh4', 'sh4eb']),
+
+        ('m68k', ['m68k']),
+        ('68020', ['m68k']),
+    ])
+
+    def __init__(self, plugin_administrator, config=None, recursive=True, unpacker=None):
+        self.unpacker = Unpacker(config) if unpacker is None else unpacker
+        super().__init__(plugin_administrator, config=config, recursive=recursive, plugin_path=__file__, timeout=600)
+
+    def process_object(self, file_object):
+        if not docker_is_running():
+            logging.error('could not process object: docker daemon not running')
+            return file_object
+
+        if self.NAME not in file_object.processed_analysis:
+            file_object.processed_analysis[self.NAME] = {}
+        file_object.processed_analysis[self.NAME]['summary'] = []
+
+        if file_object.processed_analysis['file_type']['mime'] in self.FILE_TYPES:
+            return self._process_included_binary(file_object)
+        else:
+            return self._process_container(file_object)
+
+    def _process_included_binary(self, file_object):
+        # File will get analyzed in the parent container
+        file_object.processed_analysis[self.NAME]['parent_flag'] = True
+        return file_object
+
+    def _process_container(self, file_object):
+        if len(file_object.files_included) == 0:
+            return file_object
+
+        tmp_dir = self.unpacker.unpack_fo(file_object)
+
+        if tmp_dir:
+            try:
+                self.root_path = tmp_dir.name
+                file_list = self._find_relevant_files(tmp_dir)
+                if file_list:
+                    file_object.processed_analysis[self.NAME]['files'] = {}
+                    self._process_included_files(file_list, file_object, self.root_path)
+            finally:
+                tmp_dir.cleanup()
+
+        return file_object
+
+    def _find_relevant_files(self, tmp_dir):
```

Comment:
this function is very complex. Please refactor.

you might use `pathlib` or `common_helper_files.get_all_files_in_dir` instead of os.walk